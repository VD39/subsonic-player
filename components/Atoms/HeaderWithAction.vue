<template>
  <div class="centerItems spaceBetween">
    <slot />

    <div v-if="$slots.actions" ref="actions" class="centerItems">
      <slot name="actions" />
    </div>
  </div>
</template>
