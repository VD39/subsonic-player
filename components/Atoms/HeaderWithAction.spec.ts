import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import HeaderWithAction from './HeaderWithAction.vue';

function factory(props = {}, slots = {}) {
  return mount(HeaderWithAction, {
    props: {
      ...props,
    },
    slots: {
      default: 'Default slot content.',
      ...slots,
    },
  });
}

describe('HeaderWithAction', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('when actions slot is not set', () => {
    it('does not show the actions wrapper', () => {
      expect(wrapper.find({ ref: 'actions' }).exists()).toBe(false);
    });
  });

  describe('when actions slot is set', () => {
    beforeEach(() => {
      wrapper = factory(undefined, {
        actions: 'Actions slot content.',
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not show the actions wrapper', () => {
      expect(wrapper.find({ ref: 'actions' }).exists()).toBe(true);
    });
  });
});
