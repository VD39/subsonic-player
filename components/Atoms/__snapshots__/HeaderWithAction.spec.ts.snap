// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderWithAction > matches the snapshot 1`] = `
"<div class="centerItems spaceBetween">Default slot content.
  <!--v-if-->
</div>"
`;

exports[`HeaderWithAction > when actions slot is set > matches the snapshot 1`] = `
"<div class="centerItems spaceBetween">Default slot content.<div class="centerItems">Actions slot content.</div>
</div>"
`;
